### PR TITLE
Fully integrate KSP to the lib and completely remove KAPT.

### PR DESCRIPTION
**Pull request by ivanisidrowu**
### Description
* Integrate KSP for generating parsers.
* Remove KAPT from KtRssReader completely.
* Update KotlinPoet to 1.10.2.
* Use `kotlinpoet-ksp` interop lib. https://square.github.io/kotlinpoet/interop-ksp/
* Run and pass all the instrumented and JVM unit tests.

### Todo
* Final checks before releasing version 2.2.

### Screenshots
Screen Shot 2022-04-03 at 6 51 11 PM
Screen Shot 2022-04-03 at 6 48 09 PM
Screen Shot 2022-04-03 at 6 46 26 PM
Screen Shot 2022-04-03 at 5 26 59 PM

### CODE REVIEW

**Comment by dev-weiqi** [comment]
@ivanisidrowu 💯💯💯

**Review by dev-weiqi — APPROVED** [review]

File: processor/src/main/java/tw/ktrssreader/processor/util/TypeExtensions.kt [hunk 1]
```diff
@@ -113,33 +100,16 @@ fun String.getVariableName(tag: String): String {
 
 fun String.appendTypeConversion(typeString: String): String {
     return when {
-        typeString.contains(String::class.java.simpleName, ignoreCase = true) -> this
-        typeString.contains(Integer::class.java.simpleName, ignoreCase = true) -> "$this?.toIntOrNull()"
-        typeString.contains(Boolean::class.java.simpleName, ignoreCase = true) -> "$this?.%M()"
-        typeString.contains(Long::class.java.simpleName, ignoreCase = true) -> "$this?.toLongOrNull()"
-        typeString.contains(Short::class.java.simpleName, ignoreCase = true) -> "$this?.toShortOrNull()"
+        typeString.contains(String::class.simpleName!!, ignoreCase = true) -> this
+        typeString.contains(Integer::class.simpleName!!, ignoreCase = true) -> "$this?.toIntOrNull()"
```
Comment (lush-life) [inline processor/src/main/java/tw/ktrssreader/processor/util/TypeExtensions.kt:104]: Replace with `Int`?

**Review by lush-life — CHANGES_REQUESTED** [review]

**Review by AmyPlace — APPROVED** [review]
👍👍👍

File: processor/src/main/java/tw/ktrssreader/processor/util/TypeExtensions.kt (hunk 1 above)
Comment (ivanisidrowu) [reply processor/src/main/java/tw/ktrssreader/processor/util/TypeExtensions.kt]: Yea, I use the wrong type. Thanks for the heads up! I modified it, please check it out.

**Review by lush-life — APPROVED** [review]

**Pull request merged**